packages/dataclass-editor/src/master/DataClassMasterContent.tsx

```diff
@@ -97,16 +109,53 @@ export const DataClassMasterContent = () => {
   useUpdateSelection(table);
 
   const deleteField = () => {
-    const { newData: newFields, selection } = deleteFirstSelectedRow(table, dataClass.fields);
+    const { newData: newFields, selection } = deleteAllSelectedRows(table, dataClass.fields);
     const newDataClass = structuredClone(dataClass);
     newDataClass.fields = newFields;
     setDataClass(newDataClass);
     setSelectedField(selection);
   };
 
+  const updateOrder = (moveId: string, targetId: string) => {
+    const selectedRows = table.getSelectedRowModel().flatRows.map(r => r.original.name);
+    const moveIds = selectedRows.length > 1 ? selectedRows : [dataClass.fields[parseInt(moveId)].name];
```

Comment:
Does this not work for the case where `selectedRows.length === 1` in the first place?
I don't think a different logic for these cases is required. You can also get rid of the argument `moveId` of this function.

Reply:
Unfortunately, it doesn't quite work because it's possible to have selected one row but then remain pressed on a different row and directely move that one. In this case, the selectedRows don't match, and instead, the row corresponding to the moveId should be used.
![moveNotSelected](https://github.com/user-attachments/assets/6d62df74-a70f-4c14-8540-5cf98aa0362e)